pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -243,13 +276,26 @@ public String validateSchema(FormDataMultiPart multiPart) {
   @ApiOperation(value = "Validate schema", notes = "This API returns the schema that matches the one you get "
       + "from 'GET /schema/{schemaName}'. This allows us to validate schema before apply.")
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "Successfully validated schema"),
-      @ApiResponse(code = 400, message = "Missing or invalid request body"),
-      @ApiResponse(code = 500, message = "Internal error")
+      @ApiResponse(code = 200, message = "Successfully validated schema"), @ApiResponse(code = 400, message =
+      "Missing or invalid request body"), @ApiResponse(code = 500, message = "Internal error")
   })
-  public String validateSchema(Schema schema) {
+  public String validateSchema(String schemaJsonString) {
+    Pair<Schema, Map<String, Object>> schemaAndUnrecognizedProps = null;
+    try {
+      schemaAndUnrecognizedProps = JsonUtils.stringToObjectAndUnrecognizedProperties(schemaJsonString, Schema.class);
+    } catch (Exception e) {
+      String msg = String.format("Invalid schema config json string: %s", schemaJsonString);
+      throw new ControllerApplicationException(LOGGER, msg, Response.Status.BAD_REQUEST, e);
+    }
+    Schema schema = schemaAndUnrecognizedProps.getLeft();
     validateSchemaInternal(schema);
-    return schema.toPrettyJsonString();
+    ObjectNode response = schema.toJsonObject();
+    response.set("unrecognizedProperties", JsonUtils.objectToJsonNode(schemaAndUnrecognizedProps.getRight()));
+    try {
+      return JsonUtils.objectToPrettyString(response);
```

Comment:
(minor) I think you can directly return the `ObjectNode`

Reply:
There's some formatting difference that way. This API used to return beautified json string. Keeping it same.